staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation.go

```diff
@@ -96,8 +93,15 @@ func convertJSONSchemaProps(in *apiextensions.JSONSchemaProps, out *spec.Schema)
 	if err := convertSliceOfJSONSchemaProps(&in.AnyOf, &out.AnyOf); err != nil {
 		return err
 	}
-	if err := convertJSONSchemaProps(in.Not, out.Not); err != nil {
-		return err
+
+	if in.Not != nil {
+		in, out := &in.Not, &out.Not
+		*out = new(spec.Schema)
+		if err := convertJSONSchemaProps(*in, *out); err != nil {
+			return err
+		}
+	} else {
+		out.Not = nil
```

Comment:
what about these else-clauses? didn't you want to drop them?

Reply:
Done.